<?xml version="1.0"?>
<launch>

  <arg name="SpawnVTOL1" default="True"/>
  <arg name="x1" default="5.0"/>
  <arg name="y1" default="5.0"/>
  <arg name="z1" default="0.0"/>

  <arg name="SpawnVTOL2" default="False"/>
  <arg name="x2" default="5.0"/>
  <arg name="y2" default="10.0"/>
  <arg name="z2" default="0.0"/>

  <arg name="SpawnVTOL3" default="False"/>
  <arg name="x3" default="5.0"/>
  <arg name="y3" default="15.0"/>
  <arg name="z3" default="0.0"/>

  <arg name="GazeboGUI" default="True"/>

  <!-- Gazebo sim -->
  <!-- Needed for quad and vtols -->
  <include file="$(find quadnodes)/launch/gazeboOutdoor.launch">
    <arg name="world" default="$(find quadnodes)/worlds/empty.world"/>
    <arg name="gui" default="$(arg GazeboGUI)"/>
  </include>

  <!-- Spawn UAV1 -->
  <include file="$(find vtolnodes)/launch/spawnVTOL1_SDF.launch" if="$(eval SpawnVTOL1)">
    <arg name="x" default="$(arg x1)"/>
    <arg name="y" default="$(arg y1)"/>
    <arg name="z" default="$(arg z1)"/>
  </include>

  <!-- Spawn UAV2 -->
  <include file="$(find vtolnodes)/launch/spawnVTOL2_SDF.launch" if="$(eval SpawnVTOL2)">
    <arg name="x" default="$(arg x2)"/>
    <arg name="y" default="$(arg y2)"/>
    <arg name="z" default="$(arg z2)"/>
  </include>

  <!-- Spawn UAV3 -->
  <include file="$(find vtolnodes)/launch/spawnVTOL3_SDF.launch" if="$(eval SpawnVTOL3)">
    <arg name="x" default="$(arg x3)"/>
    <arg name="y" default="$(arg y3)"/>
    <arg name="z" default="$(arg z3)"/>
  </include>


  <group ns="UAV1" if="$(eval SpawnVTOL1)">
    <!-- Motion planner -->
    <node name="mavrosVTOLWaypoints" pkg="vtolnodes" type="mavrosVTOLWaypoints.py" output="screen" >
      <rosparam param="xWaypointList"> [0, 100, 100, 0]</rosparam>
      <rosparam param="yWaypointList"> [0, 0, 100, 100]</rosparam>
      <rosparam param="zWaypointList"> [5, 5, 5, 5]</rosparam>
      <param name="waypointRadius"         value="5"       />   # m
      <param name="waypointIndex"          value="0"          />   # [-]
      <param name="spawnBiasX"             value="$(arg x1)"  />   # m
      <param name="spawnBiasY"             value="$(arg y1)"  />   # m
      <param name="spawnBiasZ"             value="$(arg z1)"  />   # m
    </node>
  </group>

  <group ns="UAV2" if="$(eval SpawnVTOL2)">
    <!-- Motion planner -->
    <node name="mavrosVTOLWaypoints" pkg="vtolnodes" type="mavrosVTOLWaypoints.py" output="screen" >
      <rosparam param="xWaypointList"> [0, 10, 10, 0]</rosparam>
      <rosparam param="yWaypointList"> [0, 0, 10, 10]</rosparam>
      <rosparam param="zWaypointList"> [10, 10, 10, 10]</rosparam>
      <param name="waypointRadius"         value="0.25"       />   # m
      <param name="waypointIndex"          value="0"          />   # [-]
      <param name="spawnBiasX"             value="$(arg x2)"  />   # m
      <param name="spawnBiasY"             value="$(arg y2)"  />   # m
      <param name="spawnBiasZ"             value="$(arg z2)"  />   # m
    </node>
  </group>

  <group ns="UAV3" if="$(eval SpawnVTOL3)">
    <!-- Motion planner -->
    <node name="mavrosVTOLWaypoints" pkg="vtolnodes" type="mavrosVTOLWaypoints.py" output="screen" >
      <rosparam param="xWaypointList"> [0, 10, 10, 0]</rosparam>
      <rosparam param="yWaypointList"> [0, 0, 10, 10]</rosparam>
      <rosparam param="zWaypointList"> [15, 15, 15, 15]</rosparam>
      <param name="waypointRadius"         value="0.25"       />   # m
      <param name="waypointIndex"          value="0"          />   # [-]
      <param name="spawnBiasX"             value="$(arg x3)"  />   # m
      <param name="spawnBiasY"             value="$(arg y3)"  />   # m
      <param name="spawnBiasZ"             value="$(arg z3)"  />   # m
    </node>
  </group>

  <!-- Create tf transformations -->
  <node pkg="tf" type="static_transform_publisher" name="UAVs2World" args="0.0 0.0 0.0 0 0 0 world worldUAVs 100" />

</launch>
